### ISSUE TITLE
Subclasses of enum.Enum can't be dumped

### ISSUE THREAD

**Issue by VOsakosa** [yaml/pyyaml#511]
Steps to reproduce:
```
import yaml
import enum
class Spam(enum.Enum):
    Value = 1
yaml.dump(Spam)
```
gives
``TypeError: __reduce_ex__() missing 1 required positional argument: 'proto'``
in ``represent_object(self, data)``.

My analysis:
The dumping code is calling ``Spam``'s ``__reduce_ex__`` _method_ on the class itself and therefore supplies one argument less than needed. Since ``enum.Enum`` has a meta class, the normal dumping for ``type`` instances is not run.

Adding a representer for ``enum.EnumMeta`` is a simple workaround, but I suppose the problem could also be solved by making
the representer for ``type`` a multi representer.

**Comment by nitzmahone** [comment]
One way or another, this shouldn't be too hard to fix for pyyaml 6.0... 

That said- I am curious what you're doing that you're serializing actual types- care to share? Serializing *instances* of custom enums makes total sense (and works fine), but the tests around serializing language infra are ancient (ie, long-predating `enum`). I'm sure there are a number of other cases like these (off the top of my head, dataclasses and namedtuples probably don't work right either), but they're also not done by the vast majority of users.

**Comment by VOsakosa** [comment]
I was dumping a quick attempt at an ECS, where types were used as keys in a dict. Essentially, a value is saved in a dict with its type as key.
I was a bit hesitant to open this issue because I knew it is an obscure case.:)

**Comment by nitzmahone** [comment]
fixed by #556
